src/js/components/DateInput/DateInput.js

```diff
@@ -92,7 +92,9 @@ const DateInput = forwardRef(
         id={inline && !format ? id : undefined}
         range={range}
         date={range ? undefined : value}
-        dates={range ? [value] : undefined}
+        // when caller initializes with empty array, dates
+        // should be undefined
```

Comment:
I found myself asking "why should this be undefined?" Suggesting we state the desired behavior "allow user to select both start and end"

```suggestion
        // when caller initializes with empty array, dates should be undefined 
        // allowing the user to select both begin and end of the range
```